src/components/Button/src/Button.vue

```diff
@@ -104,6 +97,33 @@ function setColorVariables(tokens, variant) {
 	};
 }
 
+const TEXT_STYLES = new Set([
+	'fontFamily',
+	'fontWeight',
+	'fontSize',
+	'fontStyle',
+	'textTransform',
+	'textDecoration',
+]);
+
+function kebabCase(string) {
+	return string.replace(/([\da-z])([A-Z])/g, '$1-$2').toLowerCase();
+}
+
+function formatCssStyles(theme, styles, suffix) {
+	const formattedStyles = {};
+	for (const [style, value] of Object.entries(styles)) {
+		if (TEXT_STYLES.has(style) && cssValidator(style, value)) {
```

Comment:
`cssValidator(style, value)` will always return a function which will always evaluate to true. It's meant for quickly setting cssValidator functions on Vue props. I think we need to add a new util function called `isValidCss(cssProperty, cssValue)` and use that here instead. Also, the current use is incorrect since we're passing the css properties in camelCase but for `CSS.supports` to work correctly we need to pass the css properties in kebab-case, example screenshot: 
![image](https://user-images.githubusercontent.com/7769424/201656135-806f46a4-24fb-4158-a33e-2e98b212d5cf.png)